src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -125,6 +130,7 @@ public void Dump()
             public void Clear()
             {
                 _objectsCache.Clear();
+                Metrics.CachedNodesCount = _count = 0;
```

Comment:
I'm not sure how fragile it is, but this might not be visible across threads as writes are without barriers. Probably small thing, not needed to adjust.